src/sentry/api/endpoints/organization_member_details.py

```diff
@@ -147,7 +169,40 @@ def put(self, request, organization, member_id):
         if has_sso:
             sso_enabled.send(organization=organization, sender=request.user)
 
-        return Response(status=204)
+        with transaction.atomic():
```

Comment:
This is a pretty big transaction here. Would like to see the scope narrowed a bit. It's also wrapping a transaction for a read query which may return early and never mutate.

I'd narrow the scope of this to just be the `OrganizationMemberTeam` row updates.